### PR TITLE
Travis: Cleanup configuration

### PR DESCRIPTION
**Pull request by core23**
- cleanup old symfony 2.8 and 3.0 hacks
- moved **php-cs-fixer** to travis
- **php-cs-fixers** runs once under PHP7
- **sphinx-build** runs once under PHP7
- added PHP nightly builds

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -23,9 +22,9 @@ matrix:
     - php: 5.4
       env: SYMFONY_VERSION=2.8.*
     - php: 5.6
-      env: SYMFONY_VERSION=2.8.* CS_FIXER=run
+      env: CS_FIXER=run
```
Comment (OskarStark) [inline .travis.yml]: wouldn't it be faster to run php-cs-fixer with php7

**Comment by core23** [comment]
Good point @OskarStark

**Comment by OskarStark** [comment]
we should fix this across the sonata bundles

**Comment by core23** [comment]
Refs https://github.com/sonata-project/SonataCoreBundle/pull/227

**Comment by OskarStark** [comment]
any news on this @core23

**Comment by core23** [comment]
I'm waiting for the php version discussion ;)

**Comment by OskarStark** [comment]
:smiley:

**Comment by greg0ire** [comment]
BTW, long lines can be avoided like this : https://github.com/greg0ire/enum-bundle/blob/master/.travis.yml#L19

File: .travis.yml [hunk 2]
```diff
@@ -33,11 +32,11 @@ matrix:
     - php: 5.6
       env: SYMFONY_VERSION=2.8.*
     - php: 5.5
-      env: SYMFONY_VERSION="3.0.*" ACL_VERSION="dev-master"
+      env: SYMFONY_VERSION=3.0.*
     - php: 5.6
-      env: SYMFONY_VERSION="3.0.*" ACL_VERSION="dev-master"
+      env: SYMFONY_VERSION=3.0.*
     - php: 7.0
-      env: SYMFONY_VERSION="3.0.*" ACL_VERSION="dev-master"
```
Comment (Soullivaneuh) [inline .travis.yml]: Why removing this?

**Comment by Soullivaneuh** [comment]
Two things:

* I think we can leave only one tests for SF 3, like other versions does.
* Please consider adding php `nightly` version on `allow_failures` section.

File: .travis.yml (hunk 2 above)
Comment (core23) [inline .travis.yml]: There is no reason for the dev-master version of the acl bundle

**Comment by core23** [comment]
Why should we use the ``nightly`` build of php? @Soullivaneuh

**Comment by Soullivaneuh** [comment]
Simply to see test of the next preparing version of PHP and prevent possible bug before stabilization of this one.

BTW, the same should be done for SF 3.1.

**Comment by core23** [comment]
:-1: The travis build is big enough

**Comment by greg0ire** [comment]
The allowed failures are not awaited to get the build status though, so do we really care?

File: .travis.yml (hunk 2 above)
Comment (Soullivaneuh) [inline .travis.yml]: Note sur about that, I think admin-bundle have to use it.

**Comment by Soullivaneuh** [comment]
> :-1: The travis build is big enough

This is not a matter of size. Sonata projects always test nightly build an allow_failure to not have to solve "tons of issues" when a new stable is out.

We have to keep this methodology.

BTW, you can easily remove 2 SF 3.0 tests from the matrix.

> The allowed failures are not awaited to get the build status though, so do we really care?

:+1:

File: .travis.yml (hunk 2 above)
Comment (greg0ire) [inline .travis.yml]: 2.3.7 needs it, but does not require it (2.3.8 will)

Comment (greg0ire) [inline .travis.yml]: That being said, if we require ACLs manually, maybe we should choose something other than `dev-master`

**Comment by core23** [comment]
@Soullivaneuh Which symfony version should we test against nightly php versions?

File: .travis.yml (hunk 2 above)
Comment (core23) [inline .travis.yml]: The dependency exists in ``composer.json`` with a concrete version. This should do it:
https://github.com/sonata-project/SonataAdminBundle/blob/master/composer.json#L33

**Comment by core23** [comment]
Can we use this travis config for all bundles? @greg0ire @OskarStark @greg0ire

File: .travis.yml [hunk 3]
```diff
@@ -50,14 +50,14 @@ before_script:
...
-  - if [ "$ACL_VERSION" != "" ]; then composer require "symfony/security-acl:${ACL_VERSION}" --no-update; fi;
   - travis_wait composer update --prefer-dist --no-interaction $COMPOSER_FLAGS
   - export PATH=$HOME/.local/bin:$PATH
   - pip install -r Resources/doc/requirements.txt --user `whoami`
 
 script:
  - if [ "$CS_FIXER" = "run" ]; then make cs_dry_run ; fi;
- - make test
+ - if [ "$DOCS" = "run" ]; then make docs ; fi;
+ - if [ "$SKIP_TESTS" != true ]; then make test ; fi;
```
Comment (greg0ire) [inline .travis.yml]: Maybe this could be replaced with `make $TARGET`, and have target be one of `test`, docs`,or `cs_dry_run`

Comment (greg0ire) [inline .travis.yml]: And have `$TARGET` default to test http://www.cyberciti.biz/tips/howto-setting-default-values-for-shell-variables.html

Comment (greg0ire) [inline .travis.yml]: Or even have `make` default to `make test`

Comment (core23) [inline .travis.yml]: IMO ``make cs`` is the best default value.

**Comment by core23** [comment]
ping @OskarStark @Soullivaneuh

**Comment by greg0ire** [comment]
Regarding the build : have you tried rebasing?

**Comment by core23** [comment]
Looks like there is a problem with the ``BaseWidgetTest`` for the old twig bundle @greg0ire

**Comment by greg0ire** [comment]
How come the tests pass in master?

File: .travis.yml [hunk 4]
```diff
@@ -14,18 +15,17 @@ env:
...
+      env: TARGET=docs
     - php: 5.3
-      env: SYMFONY_VERSION=2.8.*
-    - php: 5.4
-      env: SYMFONY_VERSION=2.8.*
-    - php: 5.6
-      env: SYMFONY_VERSION=2.8.* CS_FIXER=run
-    - php: 5.3
-      env: SYMFONY_VERSION=2.3.* COMPOSER_FLAGS="--prefer-lowest"
+      env: COMPOSER_FLAGS="--prefer-lowest --prefer-stable"
```
Comment (greg0ire) [inline .travis.yml]: @core23 I just noticed you added `--prefer-stable`. Maybe this is why tests are broken. Could you push a commit without it, just to see?

Comment (core23) [inline .travis.yml]: Same result :/

Comment (greg0ire) [inline .travis.yml]: That's really odd… must be something else, but what?

Comment (greg0ire) [inline .travis.yml]: Looks like your no longer forcing sf `2.3.*`, that might be the reason

Comment (core23) [inline .travis.yml]: IMO setting 2.3 and prefering the lowest stable version should do the same. That's why I want to remove the ``SYMFONY_VERSION=2.3``.

Comment (greg0ire) [inline .travis.yml]: But can you try with 2.3.* specified though? Just to see if it comes from that?

Comment (greg0ire) [inline .travis.yml]: Not sure I agree with you on that BTW, if we do not specify 2.3, it means we support things that might be incompatible with it, for no valid reason.

**Comment by greg0ire** [comment]
Ok so the solution : 

- [] remove the last commit
- [] use `symfony/form` and `symfony/twig-bridge` `^2.3.5|~3.0`

**Comment by core23** [comment]
@OskarStark Can give a final review?

**Comment by OskarStark** [comment]
looks good to me :+1:

File: composer.json [hunk 5]
```diff
@@ -18,15 +18,16 @@
     ],
     "require": {
         "symfony/http-foundation": "~2.3|~3.0",
-        "symfony/form": "~2.3|~3.0",
+        "symfony/form": "~2.3,>=2.3.5|~3.0",
```
Comment (greg0ire) [inline composer.json]: Use the caret operator as advised please

Comment (greg0ire) [inline composer.json]: So `^2.3.5`, I believe it means the same

**Pull request merged**